iothub/device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/ClientConfiguration.java

```diff
@@ -470,6 +476,15 @@ public String getGatewayHostname()
         return this.authenticationProvider.getGatewayHostname();
     }
 
+    /**
+     * Getter for MqttGatewayHostName.
+     * @return the value of MqttGatewayHostName
+     */
+    public String getMqttGatewayHostName()
```

Comment:
IIRC, this class or the authentication providers will provide the transport layer the "correct" host to connect to. The MQTT layer doesn't need to know the gateway host name + MQTT gateway host name + IoT hub host name. To simplify things, I'd like to see this be a boolean like "isConnectingToMqttGateway". See #1709

Reply:
That way the MQTT layer can do things like

```java
if (config.isConnectionToMqttGateway)
{
    topic = "..."
}
else 
{
    topic = "..."
}
```

Reply:
Added.